Tests/SwiftStdlibTests/KeyedDecodingContainerTests.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  KeyedDecodingContainerTests.swift
+//  SwifterSwift
+//
+//  Created by Francesco Deliro on 23/10/2019.
+//  Copyright © 2019 SwifterSwift
+//
+
+import XCTest
+
+@testable import SwifterSwift
+
+private struct Video: Decodable {
+    var isPlaying: Bool
+    var isFullScreen: Bool?
+
+    enum CodingKeys: String, CodingKey {
+        case isPlaying
+        case isFullScreen
+    }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        isPlaying = try container.decodeBoolAsIntOrString(forKey: .isPlaying)
+        isFullScreen = try container.decodeBoolAsIntOrStringIfPresent(forKey: .isFullScreen)
+    }
+}
+
+private enum ElementToTest: Decodable {
+    case video(Video)
+
+    enum CodingKeys: String, CodingKey {
+        case data
+    }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self = .video(try container.decode(Video.self, forKey: .data))
+    }
+}
+
+final class KeyedDecodingContainerTests: XCTestCase {
+
+    private let isPlayingAndIsFullScreenAsInt = Data("""
+    {
+      "data" : {
+        "isPlaying" : 1,
+        "isFullScreen" : 0
+      }
+    }
+    """.utf8)
+
+    private let isPlayingAndIsFullScreenAsString = Data("""
+    {
+      "data" : {
+        "isPlaying" : "true",
+        "isFullScreen" : "false"
+      }
+    }
+    """.utf8)
+
+    private let isPlayingAndIsFullScreenAsBool = Data("""
+    {
+      "data" : {
+        "isPlaying" : true,
+        "isFullScreen" : false
+      }
+    }
+    """.utf8)
+
+    private let isPlayingAsIntAndIsFullScreenNotPresent = Data("""
+    {
+      "data" : {
+        "isPlaying" : 0
+      }
+    }
+    """.utf8)
+
+    private let isPlayingAndIsFullScreenTypeMismatch = Data("""
+    {
+      "data" : {
+        "isPlaying" : [0],
+        "isFullScreen" : ["false"]
+      }
+    }
+    """.utf8)
+
+    func testDecodeBoolAsIntOrStringDataAsIntSuccessful() {
+        let result = try? JSONDecoder().decode(ElementToTest.self, from: isPlayingAndIsFullScreenAsInt)
```

Comment:
Extract this to a private function and change all the test `Data` objects above to string literals so that you're not initialising each one of them individually using their `utf8` representations.